lib/node_modules/@stdlib/array/complex64/test/test.reduce.js

```diff
@@ -0,0 +1,181 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var hasOwnProp = require( '@stdlib/assert/has-own-property' );
+var isFunction = require( '@stdlib/assert/is-function' );
+var caddf = require( '@stdlib/math/base/ops/caddf' );
+var instanceOf = require( '@stdlib/assert/instance-of' );
+var realf = require( '@stdlib/complex/realf' );
+var imagf = require( '@stdlib/complex/imagf' );
+var Complex64 = require('@stdlib/complex/float32');
+var Complex64Array = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof Complex64Array, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'attached to the prototype of the main export is a `reduce` method', function test( t ) {
+	t.strictEqual( hasOwnProp( Complex64Array.prototype, 'reduce' ), true, 'has property' );
+	t.strictEqual( isFunction( Complex64Array.prototype.reduce ), true, 'has method' );
+	t.end();
+});
+
+tape( 'the method throws an error if invoked with a `this` context which is not a complex number array instance', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 5 );
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.map.reduce( value, caddf );
+		};
+	}
+});
+
+tape( 'the method throws an error if provided a first argument which is not a function', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 10 );
+
+	values = [
+		'5',
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[]
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.reduce( value );
+		};
+	}
+});
+
+tape( 'the method throws an TypeError if provided an empty complex number array and not provided initial value', function test( t ) {
+	var arr;
+
+	arr = new Complex64Array( 0 );
+	t.throws( foo, TypeError, 'throws an error' );
+	t.end();
+
+	function foo() {
+		arr.reduce( caddf );
+	}
+});
+
+tape( 'the method takes initial value as the first element of the array if not provided', function test( t ) {
+	var expected;
+	var actual;
+	var arr;
+
+	arr = new Complex64Array( [ 1.0, -1.0, 0.0, 0.0, 0.0, 0.0 ] );
+	expected = new Complex64( 1.0, -1.0 );
+	actual = arr.reduce( caddf );
```

Comment:
@Jaysukh-409 I think we should test more explicitly here. Namely, you want to test the arguments provided to the callback. In pseudocode, for the first invocation of the callback, we should expect

```javascript
function reducer( acc, v ) {
    t.strictEqual( acc, arr[ 0 ] );
    t.strictEqual( v, arr[ 1 ] );
    ...
}
```